pkg/backup/restore_online.go

```diff
@@ -342,6 +349,40 @@ func sendAddRemoteSSTWorker(
 
 				log.Dev.VInfof(ctx, 1, "restoring span %s of file %s (file span: %s)", restoringSubspan, file.Path, file.BackupFileEntrySpan)
 				file.BackupFileEntrySpan = restoringSubspan
+
+				// swap out any external connections for their underlying uris and revalidate
+				uri, err := url.Parse(file.Dir.URI)
+				if err != nil {
+					return errors.Wrap(err, "invalid URI")
+				}
+				if uri.Scheme == externalconn.Scheme {
+					var ec externalconn.ExternalConnection
+					var ok bool
+					if ec, ok = extConnCache[uri.Host]; !ok {
+						if err := execCtx.ExecCfg().InternalDB.Txn(ctx, func(ctx context.Context, t isql.Txn) error {
+							ec, err = externalconn.LoadExternalConnection(ctx, uri.Host, t)
+							if err != nil {
+								return err
+							}
+							extConnCache[uri.Host] = ec
+							return nil
+						}); err != nil {
+							return errors.Wrap(err, "failed to load external connection")
+						}
+					}
+
+					materialized, err := externalconn.Materialize(ec, uri)
+					if err != nil {
+						return errors.Wrap(err, "failed to materialize external connection URI")
+					}
+					// revalidate
```

Comment:
nit: we should describe why we need to revalidate in the comment. Is there a specific edge case you are worried about?